fix HTML generated by the simple templates

return42 opened the pull request:
This is another PR that improves the quality of the generated HTML markup from the ``simple`` templates.  The discussion on this topic is conducted under the incident https://github.com/searxng/searxng/issues/3793

> ``/preferences``
> 
> There are too many to list out, most of which point to aria-labelledby.

In its previous implementation, the macro ``checkbox_onoff_reversed`` always created an ``aria-labelledby`` attribute, even if there was no descriptive tag with the generated ID (used as the value of the ``aria-labelledby``).

Before this patch, the Nu-HTML-Checker [1] reported 255 issues of this type::

    The aria-labelledby attribute must point to an element in the same document. (255)

[1] https://validator.w3.org/nu/

---

For more details read the commit messages in this patch series

Commit:
[fix] Removes ``/>`` ending tags for void HTML elements

continuation of commit: 0b832f19b

Signed-off-by: Markus <markus@venom.fritz.box>

Commit:
[fix] simple template: add ID to elements used for aria-labelledby

The ``aria-labelledby`` [1] attribute identifies the element that labels the
element it is applied to.  The templates ``infinite_scroll.html`` and
``search_on_category_select.html`` define a ``aria-labelledby`` at the <input>
tag but miss the id in the <div> with the description.

[1] https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby

Signed-off-by: Markus <markus@venom.fritz.box>

Commit:
[fix] simple template: macro checkbox_onoff_reversed

In its previous implementation, the macro ``checkbox_onoff_reversed`` always
created an ``aria-labelledby`` attribute, even if there was no descriptive tag
with the generated ID (used as the value of the ``aria-labelledby``).

Before this patch, the Nu-HTML-Checker [1] reported 255 issues of this type::

    The aria-labelledby attribute must point to an element in the same document. (255)

[1] https://validator.w3.org/nu/

Signed-off-by: Markus <markus@venom.fritz.box>

glanham-jr commented on searx/templates/simple/preferences/answerers.html:
```diff
@@ -33,10 +33,10 @@
         <tr>{{- '' -}}
-          <td class="checkbox-col">{{- checkbox_onoff_reversed('plugin_' + plugin.id, plugin.id not in allowed_plugins) -}}</td>{{- '' -}}
+          <td class="checkbox-col">{{- checkbox_onoff_reversed('plugin_' + plugin.id, plugin.id not in allowed_plugins, 'plugin_labelledby' + plugin.id) -}}</td>{{- '' -}}
           <td>{{ plugin.query_keywords|join(', ') }}</td>{{- '' -}}
           <td>{{ _(plugin.name) }}</td>{{- '' -}}
-          <td>{{ _(plugin.description) }}</td>{{- '' -}}
+          <td id="{{ 'plugin_labelledby' + plugin.id }}">{{ _(plugin.description) }}</td>{{- '' -}}
```
We have the same ID used in the preferences template. Is there any risk in ID collision in answers and preferences?

return42 replied:
The `id` is build up from the prefix `plugin_labelledby` plus the value from the `plugin.id` .. since IDs of plugins are unique, I don't expect a ID collision from those of *answers* and *preferences*.

glanham-jr reviewed (approved):
Just one clarification comment, but otherwise looks good to me.

Commit:
[fix] simple template: drop useless ``aria-labelledby`` attributes

This patch removes the ``aria-labelledby`` attributes for which there is no tag
with the corresponding ID.

Reported-by: @glanham-jr https://github.com/searxng/searxng/issues/3793#issuecomment-2351689483
Signed-off-by: Markus <markus@venom.fritz.box>
